src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(5)) {
+            case 0:
+                return new Square(
+                        random.nextDouble() * 10,
+                        colorSupplier.getRandomColor()
+                );
+            case 1:
+                return new Rectangle(
+                        random.nextDouble() * 10,
+                        random.nextDouble() * 10,
+                        colorSupplier.getRandomColor()
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables for the width, length, and color and pass them on for better code readability.